Update all of the docs to reflect new org service details (#244)
moderneinc/moderne-docs: docs/administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md

description: How to configure an organizational hierarchy in Moderne DX.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Configure an organizational hierarchy with Moderne DX

In order for users to effectively access and run recipes against specific groups of repositories, Moderne DX admins will need to define an organizational structure for their company. This structure, defined via a `repos.csv` file, needs to either be deployed as a static file alongside DX or be accessible from DX via an unauthenticated URI.

This guide will provide you with everything you need to know to configure either of those options.

#### Prerequisites

This guide assumes that:

* You are an admin of Moderne DX
* You have deployed Moderne DX in your environment
* You have already configured Moderne DX to connect to your [Artifactory](./configure-dx-with-artifactory-access.md) or [Nexus](./configure-dx-with-maven-repository-access.md) instance.
* You have already built a `repos.csv` file as part of the [mass ingestion step](./mass-ingest-and-run-dx.md)

## Updating the `repos.csv` file to define an organizational hierarchy

> You should already have a `repos.csv` file that you created as part of [mass ingestion](./mass-ingest-and-run-dx.md). To group those repositories into a hierarchy of organizations, you can add one or more organization columns to the end of each row.

Reviewer: I'm not sure if we want to mention here but repos.csv used during ingestion can omit branch, repos.csv for org services must specify repo/branch

Reviewer 2: I would not mention here yet — we will update the mass ingest example to output the "effective" repos.csv as a later step and that will require its own docs update.

| Variable Name                                     | Required | Default | Description                                                                                                                                                                                                                                      |
|---------------------------------------------------|----------|---------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `MODERNE_DX_ORGANIZATION_REPOSCSV`                | `true`   |         | The path of your `repos.csv` file that provides organization information. This could also be an unauthenticated HTTP/S URI in the form of `https://your-serve/repos.csv`.            |

Reviewer 2: ```suggestion
| `MODERNE_DX_ORGANIZATION_REPOSCSV`                | `true`   |         | The path of your `repos.csv` file that provides organization information. This could also be an unauthenticated HTTP/S URI in the form of `https://your-server/repos.csv`.            |
```

> | `MODERNE_DX_ORGANIZATION_URL`                     | `false`   |         | The URL of your GraphQL service.                                                                                                              |

Reviewer 2: This is going to be confusing for users. Since you said that the `reposCsv` can be an unauthenticated HTTP/S URI, they're going to be confused as to what this URL is for. 

Maybe move this to the bottom, or change it in such a way:

```suggestion
| `MODERNE_DX_ORGANIZATION_URL`                     | `false`   |         | The URL of an optional additional customization service. See [Additional customization](#additional-customization) for more information.                                                                                                              |
```

Also don't forget about `MODERNE_DX_ORGANIZATION_SKIPSSL`.

| Argument Name                                     | Required | Default | Description                                                                                                                                                                                                                                        |
|---------------------------------------------------|----------|---------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `--moderne.dx.organization.reposCsv`              | `true`   |         | The path of your `repos.csv` file that provides organization information. This could also be an unauthenticated HTTP/S URI in the form of `https://your-serve/repos.csv`.                                                                                                                   |

> | `--moderne.dx.organization.url`                   | `false`   |         | The URL of your GraphQL service.                                                                                                              |

Reviewer 2: same comment here as in the env var section.

```

Once these are both done, your developers can now start running recipes and committing the results!

## Additional customization

> You may want to create a dedicated organizations service if you want to:
> 
> * Limit access to the organizations you've defined above so that some users only have access to some repositories 
> * Customize commit messages by repository (such as adding a JIRA ticket per repository)

Reviewer 2: DX doesn't have the ability to customize commit messages by repository, so I would suggest we simplify to this:

```suggestion
You may want to create a dedicated organizations service if you want to limit access to the organizations you've defined above so that some users only have access to some repositories.
```
